cloud/cloud_env_impl.cc

```diff
@@ -2053,27 +2054,60 @@ Status CloudEnvImpl::CheckValidity() const {
   }
 }
 
-Status CloudEnvImpl::FindAllLiveFiles(const std::string& bucket,
-                                      const std::string& object_path,
+Status CloudEnvImpl::FetchCurrentManifest(const std::string& local_dbname) const {
+  assert(cloud_manifest_);
+  auto current_epoch = cloud_manifest_->GetCurrentEpoch().ToString();
+  if (HasDestBucket()) {
+    Status st = GetStorageProvider()->GetCloudObject(
+      GetDestBucketName(), ManifestFileWithEpoch(GetDestObjectPath(), current_epoch),
+      ManifestFileWithEpoch(local_dbname, current_epoch));
+    if (!st.ok() && !st.IsPathNotFound()) {
+      return st;
+    }
+
+    if (st.ok()) {
+      return st;
+    }
+  }
+
+  // we couldn't get cloud manifest from dest, need to try from src
+  if (HasSrcBucket() && !SrcMatchesDest()) {
+    Status st = GetStorageProvider()->GetCloudObject(
+      GetSrcBucketName(), ManifestFileWithEpoch(GetSrcObjectPath(), current_epoch),
+      ManifestFileWithEpoch(local_dbname, current_epoch));
+    if (!st.ok() && !st.IsPathNotFound()) {
+      return st;
+    }
+
+    if (st.ok()) {
+      return st;
+    }
+  }
+
+  return Status::NotFound("Manifest file with epoch: " + current_epoch +
+                          " is not found in s3");
+}
```

Comment:
I'm pretty sure you don't need this and can just do `NewSequentialFile()` with a manifest filename, which will do the right thing: https://github.com/rockset/rocksdb-cloud/blob/cb15466d6d7b9d070743e4612aae8777eb94dca1/cloud/cloud_env_impl.cc#L171-L180